crates/core/tedge_agent/src/agent.rs

```diff
@@ -82,7 +84,7 @@ impl AgentConfig {
         let tedge_config = config_repository.load()?;
 
         let config_dir = tedge_config_location.tedge_config_root_path.clone();
-        let tmp_dir = tedge_config.tmp.path.clone();
+        let tmp_dir = Arc::from(tedge_config.tmp.path.as_path());
```

Comment:
Do we need to use `.as_path` here, could we not just pass `&tedge_config.tmp.path`?

Reply:
Yes, `&tedge_config.tmp.path` resolves to `&Utf8PathBuf` instead of `&Utf8Path`